vendor/engines/bulk_email/app/assets/javascripts/bulk_email/bulk_email.js.coffee

```diff
@@ -9,6 +9,23 @@ class window.BulkEmailSearchForm
 
   selectedUserTypes: -> @$userTypeCheckboxes().filter(':checked').map -> @.value
 
+  dateIrrelevantSelectedOnly: ->
+    user_types = @selectedUserTypes().toArray()
+    (user_types.includes('authorized_users') || user_types.includes('training_requested')) &&
```

Comment:
This is kind of a confusing and very verbose check.

What if we add a data attributes to these fields  marking them as date relevant/irrelevant and then look to make sure our logic passes? I'd prefer if we can avoid that mapping in the JS if we can.